### PR TITLE
Add distributed tracing support for the Symfony Cache component

### PR DESCRIPTION
**Pull request by ste93cry**
Here we go with another instrumentation, this time for the [Cache](https://symfony.com/doc/current/components/cache.html) component. Its implementation is quite simple, and it resembles the one of the DBAL.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 - Add support for distributed tracing of Twig template rendering (#430)
 - Add support for distributed tracing of SQL queries while using Doctrine DBAL (#426)
 - Add support for distributed tracing when running a console command (#455)
-- Added missing `capture-soft-fails` config schema option (#417)
```
Comment (Jean85) [inline CHANGELOG.md:9]: Is this removal intentional or has it been lost during rebase?

File: tests/DependencyInjection/ConfigurationTest.php [hunk 2]
```diff
@@ -46,6 +47,9 @@ public function testProcessConfigurationWithDefaultConfiguration(): void
                 'twig' => [
                     'enabled' => class_exists(TwigBundle::class),
                 ],
+                'cache' => [
+                    'enabled' => interface_exists(CacheInterface::class),
```
Comment (Jean85) [inline tests/DependencyInjection/ConfigurationTest.php:51]: This needs to be added to the `full` configuration fixture.

**Review by Jean85 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ste93cry) [reply CHANGELOG.md:9]: It's a duplicate entry (fix has been released in `4.0.1`), probably the result of a bad merge between `master` and `develop`

File: tests/DependencyInjection/ConfigurationTest.php (hunk 2 above)
Comment (ste93cry) [reply tests/DependencyInjection/ConfigurationTest.php:51]: Does it makes sense? Unlike the other instrumentations, this one does not register any service by default so there is nothing to test, apart from the compiler pass

Comment (Jean85) [reply tests/DependencyInjection/ConfigurationTest.php:51]: Services get still decorated; so it makes sense to test that somehow in end2end.

Comment (ste93cry) [reply tests/DependencyInjection/ConfigurationTest.php:51]: The decoration is tested in the `CompilerPassTest` class, it's kinda of an integration test as it really uses the `ContainerBuilder` to execute all steps 😉 The only thing that is mocked is the service definition of the abstract services

**Review by Jean85 — APPROVED** [review]

**Comment by Jean85** [comment]
Build should be fixed by #480

**Comment by VincentLanglet** [comment]
Hi @ste93cry  Could some documentation of this option be added in the Readme ?
I don't understand what it does.

**Comment by ste93cry** [comment]
Please open an issue in [`getsentry/sentry-docs`](https://github.com/getsentry/sentry-docs), documentation indeed needs some updating

**Comment by VincentLanglet** [comment]
Done https://github.com/getsentry/sentry-docs/issues/3463

**Pull request merged**